### ISSUE TITLE
Clearity slider

### ISSUE THREAD

**Issue by masc4ii** [ilia3101/MLV-App#98]
Playing around I found a solution which brings an effect very similar to adobes clearity slider.

Original:
![m23-1255_frame_1](https://user-images.githubusercontent.com/30245296/43284017-da2b4290-911a-11e8-860e-60805923a2a4.png)

Full positive:
![m23-1255_frame_1clearity](https://user-images.githubusercontent.com/30245296/43284064-fc194578-911a-11e8-811c-78d89a05b835.png)

Full negativ:
![m23-1255_frame_1noclearity](https://user-images.githubusercontent.com/30245296/43284073-04bb4546-911b-11e8-935c-354ba8084d32.png)

Do you like it? You like to have it? @ilia3101 : or shall I better stop with messing your code?!

**Comment by ilia3101** [comment]
I do like this!

Keep messing all you like :wink:

What kind of blurring filter is it using for the contrast? It looks like normal kind of blur maybe gaussian, I think it would look even better if it used something like bilateral or this: https://sites.fas.harvard.edu/~cs278/papers/l0.pdf, but this is only if the world was perfect (too much time to implement).

Also what if you make larger radius?

**Comment by ilia3101** [comment]
My only 'rule' for 'messing' with the code: Put an if statement, so if the filter is set to zero no cpu is wasted :wink:

**Comment by masc4ii** [comment]
You like to try it out yourself? Compile latest version and for full positive:
 set Contrast to 100, Highlights to -100 and Shadows to 100. For full negative do it the other way. You got it? All I would do is just a kind of copy of your shadow/highlight algorithm and my/your contrast, syncing the internal values and voilà. 😛

**Comment by masc4ii** [comment]
>My only 'rule' for 'messing' with the code: Put an if statement, so if the filter is set to zero no cpu is wasted 😉 (I'm sure you do)

Yes... that is right, and I will also do this for Vibrance, maybe also for saturation.

**Comment by masc4ii** [comment]
Now you can try out with latest release. The slider is there! Have fun!

**Comment by bouncyball-git** [comment]
Very nice! Did you implemented some local micro contrast algo? It even does dehase at some extend.

**Comment by masc4ii** [comment]
It results in a kind of local micro contrast algo. I lower the contrast on a blurred copy and blend it with a picture with raised contrast. This brings in the blur-radius more/less micro contrast.
When combining this with contrast + highlights/shadows pictures can look very bad/wrong. This is the price to pay using this method...

**Comment by bouncyball-git** [comment]
Understood. Will try to reproduce.

**Comment by masc4ii** [comment]
![bildschirmfoto 2018-07-30 um 20 28 48](https://user-images.githubusercontent.com/30245296/43415923-3292394c-9437-11e8-9771-e9c84c28f6ed.png)
Modern art... 😜
